src/segment-loader.js

```diff
@@ -1505,9 +1458,40 @@ export default class SegmentLoader extends videojs.EventTarget {
       part,
       byteLength: 0,
       transmuxer: this.transmuxer_,
-      audioAppendStart,
-      gopsToAlignWith
+      // type of getMediaInfoForTime that was used to get this segment
+      getMediaInfoForTime
     };
+
+    if (!noTimestampOffset) {
+      segmentInfo.timestampOffset = timestampOffsetForSegment({
+        segmentTimeline: segment.timeline,
+        currentTimeline: this.currentTimeline_,
+        startOfSegment,
+        buffered: this.buffered_(),
+        overrideCheck: this.isPendingTimestampOffset_
+      });
+      this.isPendingTimestampOffset_ = false;
+    }
+
+    const audioBufferedEnd = lastBufferedEnd(this.sourceUpdater_.audioBuffered());
+
+    if (typeof audioBufferedEnd === 'number') {
```

Comment:
why did the logic for this part change?

Reply:
ah, I see. `lastBufferedEnd` already does a null check and so if we don't get a number from it, we don't have audio buffered but if we do we can just use that value directly in the next step.